x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -2652,6 +2652,52 @@ export class RulesClient {
     }
   }
 
+  public async runSoon({ id }: { id: string }) {
+    const { attributes } = await this.unsecuredSavedObjectsClient.get<Rule>('alert', id);
+    const taskDoc = await this.taskManager.get(id);
```

Comment:
I would move this after the authorization check so if a user doesn't have access to the rule, we don't do an unnecessary get for the task document.

Reply:
Resolved in https://github.com/elastic/kibana/pull/139848/commits/8d739ac9c783f035e6dba8bd3dfe1b20425b173f